content/300-guides/300-upgrade-guides/200-upgrading-to-latest/10-upgrade-to-2270.mdx

```diff
@@ -0,0 +1,150 @@
+
+### Upgrade paths for Prisma Migrate users
+
+Use the following guide if your project uses Prisma Migrate. Choose the upgrade path that matches your requirements:
+
+* [I want to maintain my existing constraint and index names](#upgrade-path-1-i-want-to-maintain-my-existing-constraint-and-index-names)
+* [I want to use Prisma's default constraint and index namess](#upgrade-path-2-i-want-to-use-prismas-default-constraint-and-index-namess)
+
+#### Upgrade path 1: I want to maintain my existing constraint and index names
+
+1. Enable the `namedConstraints` Preview:
+
+    ```prisma highlight=3;normal
+    generator client {
+      provider        = "prisma-client-js"
+      previewFeatures = ["namedConstraints"]
+    }
+
+    datasource db {
+      provider = "postgresql"
+      url      = env("DATABASE_URL")
+    }
+
+    model User {
+      id    Int    @id() @default(autoincrement())
+      name  String @unique
+      posts Post[]
+    }
+
+    model Post {
+      id          Int     @id @default(autoincrement())
+      title       String
+      authorName  String  @default("Anonymous")
+      author      User?   @relation(fields: [authorName], references: [name])
+    }
+    ```
+   
+2. Introspect your database to populate the Prisma schema with constraint and index names in your underlying database _that do not match Prisma's naming convention_:
+
+   ```terminal
+   npx prisma migrate introspect
+   ```
+
+   In this example, the highlighted constraints did not conform to Prisma's default naming convention and now include the `map` attribute field:
+
+   ```prisma
+   model User {
+     id    Int    @id(map: "Custom_Constraint_Name") @default(autoincrement())
+     name  String @unique
+     posts Post[]
+   }
+
+   model Post {
+     id    Int @id @default(autoincrement())
+     title String
+     authorName String @default("Anonymous")
+     author User? @relation(fields: [authorName], references: [name], map: "Custom_Foreign_Key_Constraint")
+   }
+   ```
+
+3. Optionally rename any explicitly defined constraint names where you do not want to use Prisma's naming convention, but do want to use a different constraint name:
+
+   ```prisma highlight=2;delete|3;add
+   model User {
+     id    Int    @id(map: "Custom_Constraint_Name") @default(autoincrement())       
+     id    Int    @id(map: "User_ID_Primary_Key") @default(autoincrement())
+     name  String @unique
+     posts Post[]
+   }
+   ```
+
+4. Optionally _remove_ the `map` field from any constraint or index attribute where you want to use the Prisma default:
+   
+   ```prisma highlight=5;delete|6;add
+   model Post {
+     id    Int @id @default(autoincrement())
+     title String
+     authorName String @default("Anonymous")
+     author User? @relation(fields: [authorName], references: [name], map: "Custom_Foreign_Key_Constraint")
+     author User? @relation(fields: [authorName], references: [name])
+   }
+   ```
```

Comment:
They could modify the introspected values to match whatever they now want?
Maybe these three (currently 2) steps can be rolled into one?

Reply:
I guess so - I separated them so that you can address each choice separately (why would I remove, why would I rename)